src/pages/WorkspaceSwitcherPage.tsx

```diff
@@ -0,0 +1,307 @@
+    if (workspace1.policyID === selectedWorkspaceID) {
+        return -1;
+    }
+    if (workspace2.policyID === selectedWorkspaceID) {
+        return 1;
+    }
+    return workspace1.text?.toLowerCase().localeCompare(workspace2.text?.toLowerCase() ?? '') ?? 0;
+};
+
+type WorkspaceSwitcherPageOnyxProps = {
+    /** The list of this user's policies */
+    policies: OnyxCollection<Policy>;
+};
+
+type WorkspaceSwitcherPageProps = WorkspaceSwitcherPageOnyxProps;
+
+function WorkspaceSwitcherPage({policies}: WorkspaceSwitcherPageProps) {
+    const theme = useTheme();
+    const styles = useThemeStyles();
+    const {isOffline} = useNetwork();
+    const [selectedOption, setSelectedOption] = useState<SimpleWorkspaceItem>();
+    const [searchTerm, setSearchTerm] = useState('');
+    const {inputCallbackRef} = useAutoFocusInput();
+    const {translate} = useLocalize();
+    const {activeWorkspaceID, setActiveWorkspaceID} = useActiveWorkspace();
+
+    const brickRoadsForPolicies = useMemo(() => getWorkspacesBrickRoads(), []);
+    const unreadStatusesForPolicies = useMemo(() => getWorkspacesUnreadStatuses(), []);
+
+    const getIndicatorTypeForPolicy = useCallback(
+        (policyId?: string) => {
+            if (policyId && policyId !== activeWorkspaceID) {
+                return brickRoadsForPolicies[policyId];
+            }
+
+            if (Object.values(brickRoadsForPolicies).includes(CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR)) {
+                return CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR;
+            }
+
+            if (Object.values(brickRoadsForPolicies).includes(CONST.BRICK_ROAD_INDICATOR_STATUS.INFO)) {
+                return CONST.BRICK_ROAD_INDICATOR_STATUS.INFO;
+            }
+
+            return undefined;
+        },
+        [activeWorkspaceID, brickRoadsForPolicies],
+    );
+
+    const hasUnreadData = useCallback(
+        // TO DO: Implement checking if policy has some unread data
+        (policyId?: string) => {
+            if (policyId) {
+                return unreadStatusesForPolicies[policyId];
+            }
+
+            return Object.values(unreadStatusesForPolicies).some((status) => status);
+        },
+        [unreadStatusesForPolicies],
+    );
+
+    const selectPolicy = useCallback(
+        (option?: SimpleWorkspaceItem) => {
+            if (!option) {
+                return;
+            }
+
+            const {policyID, isPolicyAdmin} = option;
+
+            if (policyID) {
+                setSelectedOption(option);
+            } else {
+                setSelectedOption(undefined);
+            }
+            setActiveWorkspaceID(policyID);
+            Navigation.goBack();
+            if (policyID !== activeWorkspaceID) {
+                Navigation.navigateWithSwitchPolicyID({policyID, isPolicyAdmin});
+            }
+        },
+        [activeWorkspaceID, setActiveWorkspaceID],
+    );
```

Comment:
On double click this callback is called twice. After the first click (and after calling`setActiveWorkspaceID`), the order of the workspaces change making the second click hit a different workspace. Resulting in unexpected workspace selection (user lands on the previous selected workspace).

First click:
<img width="525" alt="Screenshot 2024-10-29 at 9 11 53 AM" src="https://github.com/user-attachments/assets/c32164b7-47a7-48c9-b0a7-7489640c90bd">


Second click:
 <img width="525" alt="Screenshot 2024-10-29 at 9 11 55 AM" src="https://github.com/user-attachments/assets/cb8b5a2e-98eb-4987-b7f5-3bc3ed13f020">

Coming from https://github.com/Expensify/App/issues/51402